### ISSUE TITLE
Versioning issue and questions on permissions

### ISSUE THREAD

**Issue by IzzySoft** [sndnv/stasis#45]
The update checker today reported multiple problems here after fetching the latest release:

```
! Binary files repo/stasis.client.android_1.apk and repo/stasis.client.android_1.1.2.apk differ
! repo/stasis.client.android_1.apk declares sensitive permission(s): android.permission.READ_EXTERNAL_STORAGE android.permission.MANAGE_EXTERNAL_STORAGE
! repo/stasis.client.android_1.apk contains signature block blobs: 0x504b4453  (DEPENDENCY_INFO_BLOCK; GOOGLE)
```

The first message means you forgot to [increase `versionCode`](https://github.com/sndnv/stasis/blob/master/client-android/app/build.gradle.kts#L67) when making the release. That's what Android uses internally to tell versions apart – so this update will reach no existing installation as each client assumes already having it (`1 == 1`).

The second line means I'll have to add those permissions to your app's "green list": a backup app of course needs full storage access (done; see screenshot below).

As for `DEPENDENCY_INFO_BLOCK`, this can easily be avoided with a minor addition to your `build.gradle`:

```gradle
android {
    dependenciesInfo {
        // Disables dependency metadata when building APKs.
        includeInApk = false
        // Disables dependency metadata when building Android App Bundles.
        includeInBundle = false
    }
}
```

For some background: that BLOB is supposed to be just a binary representation of your app's dependency tree. But as it's encrypted with a public key belonging to Google, only Google can read it – and nobody else can even verify what it really contains. More details can be found e.g. here: [Ramping up security: additional APK checks are in place with the IzzyOnDroid repo](https://android.izzysoft.de/articles/named/iod-scan-apkchecks#blobs).

That said: you might wish to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to [your app at the IzzyOnDroid repo](https://apt.izzysoft.de/packages/stasis.client.android) e.g. from your Readme. And congrats, your app was just confirmed as [Reproducible Build](https://reproducible-builds.org/) (also see [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)) and thus will carry the green shield indicating that:

![image](https://github.com/user-attachments/assets/5d84c8ce-f4aa-484c-a8ca-c94bfc0cec8b)

**Comment by IzzySoft** [comment]
PS: forgot to link [your welcome toot](https://floss.social/@IzzyOnDroid/113047128086507963):
![image](https://github.com/user-attachments/assets/19b1ff05-0b31-4c52-a8d9-d4276ccde494)

**Comment by sndnv** [comment]
Thanks for reporting this! :thinking:  I'll have a better look in the following days/weeks.

**Comment by IzzySoft** [comment]
Thanks! If you have any questions, just ask :smiley:

**Comment by sndnv** [comment]
Should be fixed in [v1.1.3](https://github.com/sndnv/stasis/releases/tag/v1.1.3) :) Let me know if that's indeed the case so I can close the issue.

**Comment by IzzySoft** [comment]
Confirmed @sndnv – as you can see:

![image](https://github.com/user-attachments/assets/aaf4c8e6-561d-4d8f-8a3e-2da1e0469d35)

the "Signing block blobs" are gone, no permissions are "chocolate" (warning color) – and the green shield is still up :smiley: Thanks a lot, and once more welcome aboard!
